modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/distruptor/inbound/InboundTransactionEvent.java

```diff
@@ -85,6 +112,13 @@ public InboundTransactionEvent(MessageStore.AndesTransaction transaction, Inboun
      * @throws AndesException
      */
     public void commit() throws AndesException {
+
+        if(currentBatchSize > maxBatchSize) {
+            currentBatchSize = 0;
+            throw new AndesException("Current enqueued batch size exceeds maximum transactional batch size of " +
+            maxBatchSize + " bytes." );
+        }
+
         if (log.isDebugEnabled()) {
             log.debug("Prepare for commit");
```

Comment:
Got your point. At the moment we don't have that ability to track publisher information (something like publisher id). What we can do is have the internal Andes channel id printed here. So that we can distinguish the publisers. But still we won't be able to pin point the publishers.